lib/hope-tabs.js

```diff
@@ -0,0 +1,722 @@
+      var feedbacks = [];
+
+      feedbacks.push(
+        scheduler.transition(() => {
+          tab.classList.add('shrink-middle');
+        }, tab, 'animationend')
+
+        .then(() => {
+          tab.classList.remove('shrink-middle');
+        })
+      );
+
+      if (top) {
+        feedbacks.push(
+          scheduler.transition(() => {
+            top.classList.add('shrink-top');
+          }, top, 'animationend')
+
+          .then(() => {
+            top.classList.remove('shrink-top');
+          })
+        );
+      }
+
+      if (bottom) {
+        feedbacks.push(
+          scheduler.transition(() => {
+            bottom.classList.add('shrink-bottom');
+          }, bottom, 'animationend')
+
+          .then(() => {
+            bottom.classList.remove('shrink-bottom');
+          })
+        );
+      }
+
+      this.els.tabs.splice(tabIndex, 1);
+      this.els.tabs.unshift(tab);
+
+      return Promise.all(feedbacks);
+    })
+
+    .then(() => {
+      var top = parseInt(tab.style.top) + GUTTER_HEIGHT;
+      var translate = - (top - this.getScrollTop());
+      var motions = [];
+
+      // Slide the selected tab up
+      // to the top of the viewport
+      motions.push(scheduler.transition(() => {
+        debug('slide up', translate);
+        tab.style.transform = `translateY(${translate}px)`;
+      }, tab, 'transitionend'));
+
+      // Slide tabs above the selected
+      // tab up the same amount so that
+      // they appear to slide together
+      toMoveUp.forEach(el => {
+        motions.push(scheduler.transition(() => {
+          el.style.transform = `translateY(${translate}px)`;
+        }, el, 'transitionend'));
+      });
+
+      // Slide all tabs after the selected
+      // tab down out of the viewport
+      tabsAfter.forEach(el => {
+        motions.push(scheduler.transition(() => {
+          el.classList.add('hide-down');
+        }, el, 'animationend'));
+      });
+
+      return Promise.all(motions);
+    })
+
+    .then(() => {
+      return scheduler.mutation(() => {
+        tab.style.removeProperty('transition');
+        tab.style.removeProperty('transform');
+
+        toMoveUp.forEach(tab => {
+          tab.style.removeProperty('transition');
+          tab.style.removeProperty('transform');
+        });
+
+        this.scrollTo(0, { instant: true });
+        return this.layout();
+      });
+    })
+
+    .then(() => {
+      this.selecting = false;
+    });
+  },
+
+  create(data) {
+    debug('create', data);
+    var li = document.createElement('li');
+    var tab = document.createElement('hope-tab');
+
+    tab.themeColor = data.themeColor;
+    tab.title = data.title;
+    tab.url = data.url;
+
+    li.appendChild(tab);
+    return li;
+  },
+
+  showFirstTab() {
+    debug('show first tab');
+    var snapHeight = this.getSnapHeight();
+    var scrollTop = this.getScrollTop();
+
+    if (scrollTop < snapHeight) return Promise.resolve();
```

Comment:
don't know if the issue was introduced by this PR but apparently we want a `<=` here for this to work properly